hypothesis-python/tests/cover/test_slices.py

```diff
@@ -0,0 +1,95 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2019 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import absolute_import, division, print_function
+
+from hypothesis import given, settings, strategies as st
+from tests.common.debug import assert_all_examples, find_any, minimal
+
+
+def test_stop_stays_within_bounds():
+    size = 1000
```

Comment:
Let's try a few different size.  `@given` might be too expensive here, but a `parametrize` would be OK:

```python
use_several_sizes = pytest.mark.parametrize("size", [1, 2, 5, 10, 100, 1000])

@use_several_sizes
def test_stop_stays_within_bounds(size):
```

and the decorator can be reused for all the `assert_all_examples` tests :smile: